File: backend/app/schemas/propagation.py
"""Schemas for Network Propagation API."""

from typing import List, Optional
from pydantic import BaseModel
from backend.app.schemas.common import PaginationMeta, ProvenanceEnum


class PropagationRecord(BaseModel):
    """Network congestion propagation or downstream starvation event."""
    timestamp: str
    seed_segment_id: str
    propagated_segment_id: str
    direction: str
    hops: int
    horizon_min: int
    propagation_score: float
    risk_level: str
    evidence_reason: str
    provenance: str = ProvenanceEnum.DERIVED.value


class PropagationListResponse(BaseModel):
    """List of propagation footprint records."""
    meta: PaginationMeta
    data: List[PropagationRecord]

File: frontend/types/propagation.ts
import { PaginationMeta } from "./api";

export interface PropagationRecord {
  timestamp: string;
  seed_segment_id: string;
  propagated_segment_id: string;
  direction: string;
  hops: number;
  horizon_min: number;
  propagation_score: number;
  risk_level: string;
  evidence_reason: string;
  provenance: string;
}

export interface PropagationListResponse {
  meta: PaginationMeta;
  data: PropagationRecord[];
}

export interface PropagationQueryParams {
  limit?: number;
  offset?: number;
  seed_segment_id?: string;
  risk_level?: string;
  direction?: string;
}
